int8 flag
---
bool success
